src/util/sock.cpp

```diff
@@ -71,6 +71,21 @@ int Sock::Connect(const sockaddr* addr, socklen_t addr_len) const
     return connect(m_socket, addr, addr_len);
 }
 
+std::unique_ptr<Sock> Sock::Accept(sockaddr* addr, socklen_t* addr_len) const
+{
+#ifdef WIN32
+    static constexpr auto ERR = INVALID_SOCKET;
+#else
+    static constexpr auto ERR = SOCKET_ERROR;
+#endif
+
+    const auto socket = accept(m_socket, addr, addr_len);
+    if (socket == ERR) {
+        return nullptr;
+    }
+    return std::make_unique<Sock>(socket);
```

Comment:
11e59842e30ab5e sub-atomic style nit, a single `return` can be nice when it's simpler and shorter. Feel free to ignore.
```diff
-    if (socket == ERR) {
-        return nullptr;
-    }
-    return std::make_unique<Sock>(socket);
+    return socket == ERR ? nullptr : std::make_unique<Sock>(socket);
```